### PR TITLE
[17827] Fix text content when editing and reentering chat

### LINKED ISSUE #17827: Wrong text when editing message after unsent message in composer and re-entering chat
# Bug Report

## Problem 
[comment]: # (A problem description)

When composing a message in a chat and starting editing another message (without deleting/sending the previous one) and leaving then re-entering the chat (e.g. back, shell), we're still editing a message, but instead of the one we started editing we have the original one we were composing. Pressing the send button will replace the edited message with the original one we wrote.

https://github.com/status-im/status-mobile/assets/21865759/93775997-9d50-4250-bdc0-c03e3c818a10

#### Expected behavior

When returning to the chat, we're editing the message we started editing

#### Actual behavior

When returning to the chat, we're editing the message we were writing before

### Reproduction

1. Open chat
2. Write a message (don't send)
3. Start editing another message (don't send)
4. Go back (or to the shell)
5. Re-enter chat
6. Wrong message is in the composer

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #17827

### Summary

Fixes an issue when users types some text in composer, starts editing a message and reopens chat

status: ready

### CODE REVIEW

**Review by mohsen-ghafouri — APPROVED** [review]
LGTM :)

**Comment by OmarBasem** [comment]
@clauxx I think you were working on something related on your PR?

File: src/status_im/chat/models/input.cljs [hunk 1]
```diff
@@ -107,14 +107,15 @@
...
-                              :sending-image))
-     :dispatch [:mention/to-input-field text current-chat-id]}))
+    {:db         (-> db
+                     (assoc-in [:chat/inputs current-chat-id :metadata :editing-message]
+                               message)
+                     (assoc-in [:chat/inputs current-chat-id :metadata :responding-to-message] nil)
+                     (update-in [:chat/inputs current-chat-id :metadata]
+                                dissoc
+                                :sending-image))
+     :dispatch-n [[:chat.ui/set-chat-input-text nil current-chat-id]
```
Comment (qfrank) [inline src/status_im/chat/models/input.cljs:117]: can i ask for an explanation here? thanks in advance @ibrkhalil

File: src/status_im2/contexts/chat/composer/effects.cljs [hunk 2]
```diff
@@ -109,18 +109,15 @@
...
          ;; A small setTimeout is necessary to ensure the statement is enqueued and will get executed
          ;; ASAP.
          ;; https://github.com/software-mansion/react-native-screens/issues/472
          (js/setTimeout #(.focus ^js @input-ref) 250)
          (.setNativeProps ^js @input-ref (clj->js {:text edit-text}))
          (reset! text-value edit-text)
-         (reset! saved-cursor-position (if (zero? text-value-count)
-                                         (count edit-text)
-                                         text-value-count)))))
+         (reset! saved-cursor-position (count edit-text)))))
```
Comment (clauxx) [inline src/status_im2/contexts/chat/composer/effects.cljs:120]: Tested and all works! Wondering why this change was needed

Comment (ibrkhalil) [reply src/status_im2/contexts/chat/composer/effects.cljs:120]: We keep both values the editing value and composer value before editing
It wasn't really necessary, Because if the text-value-count existed while editing the cursor would move to the text-value-count rather than the edit-text-count.

File: src/status_im/chat/models/input.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/status_im/chat/models/input.cljs:117]: We don't need the value of the text input before editing if we initiated the editing.
Keeping It in app-db causes issues.

**Review by clauxx — APPROVED** [review]

**Review by qfrank — APPROVED** [review]

**Comment by churik** [comment]
blocking until https://github.com/status-im/status-mobile/pull/17785 merge - should be rebased and rechecked after, as composer behavior drastically changed there

**Comment by churik** [comment]
@ibrkhalil 
is it not a part of https://github.com/status-im/status-mobile/pull/17785 ?
If it is, we can close it, thank you!

If it is not - we can rebase it and recheck.
